programs/svm-spoke/src/instructions/deposit.rs

```diff
@@ -156,3 +211,53 @@ pub fn deposit_v3_now(
 
     Ok(())
 }
+
+pub fn unsafe_deposit_v3(
+    ctx: Context<DepositV3>,
+    depositor: Pubkey,
+    recipient: Pubkey,
+    input_token: Pubkey,
+    output_token: Pubkey,
+    input_amount: u64,
+    output_amount: u64,
+    destination_chain_id: u64,
+    exclusive_relayer: Pubkey,
+    deposit_nonce: u64,
+    quote_timestamp: u32,
+    fill_deadline: u32,
+    exclusivity_parameter: u32,
+    message: Vec<u8>,
+) -> Result<()> {
+    // Calculate the unsafe deposit ID as a [u8; 32]
+    let deposit_id = get_unsafe_deposit_id(ctx.accounts.signer.key(), depositor, deposit_nonce);
+    _deposit_v3(
+        ctx,
+        depositor,
+        recipient,
+        input_token,
+        output_token,
+        input_amount,
+        output_amount,
+        destination_chain_id,
+        exclusive_relayer,
+        deposit_id,
+        quote_timestamp,
+        fill_deadline,
+        exclusivity_parameter,
+        message,
+    )?;
+
+    Ok(())
+}
+
+// Define a dummy context struct so we can export this as a view function in lib.
+#[derive(Accounts)]
+pub struct Null {}
+pub fn get_unsafe_deposit_id(msg_sender: Pubkey, depositor: Pubkey, deposit_nonce: u64) -> [u8; 32] {
```

Comment:
Do you think it makes sense to expose this? Are we planning to use it off-chain, or should we implement a similar function in TS for efficiency?

Reply:
yes, I think having it exposed is useful. it's done in solidity and it's good to not force users to have to do this in js. for example say you wanted to do the deposit with etherscan for example. i don't think there is any downside in having it public. do you agree?